java/statestore/src/test/java/sleeper/statestore/s3/S3StateStoreIT.java

```diff
@@ -981,179 +986,64 @@ public void shouldReturnAllPartitions() throws IOException, StateStoreException
         // Given
         Field field = new Field("key", new LongType());
         Schema schema = Schema.builder().rowKeyFields(field).build();
-        Region region0 = new Region(new RangeFactory(schema).createRange(field, Long.MIN_VALUE, 1L));
-        Partition partition0 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region0)
-                .id("id0")
-                .leafPartition(true)
-                .parentPartitionId("root")
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region1 = new Region(new RangeFactory(schema).createRange(field, 1L, 100L));
-        Partition partition1 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region1)
-                .id("id1")
-                .leafPartition(true)
-                .parentPartitionId("root")
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region2 = new Region(new RangeFactory(schema).createRange(field, 100L, 200L));
-        Partition partition2 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region2)
-                .id("id2")
-                .leafPartition(true)
-                .parentPartitionId("root")
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region3 = new Region(new RangeFactory(schema).createRange(field, 200L, null));
-        Partition partition3 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region3)
-                .id("id3")
-                .leafPartition(true)
-                .parentPartitionId("root")
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        StateStore dynamoDBStateStore = getStateStore(schema, Arrays.asList(partition0, partition1, partition2, partition3));
 
+        PartitionTree tree = new PartitionsBuilder(schema)
+                .rootFirst("root")
+                .splitToNewChildren("root", "left", "right", 100L)
+                .splitToNewChildren("left", "id0", "id2", 1L)
+                .splitToNewChildren("right", "id1", "id3", 200L)
+                .buildTree();
+        StateStore stateStore = inMemoryStateStoreWithPartitions(tree.getAllPartitions());
```

Comment:
The code under test is getStateStore(...).getAllPartitions(), which should be called here. At the moment it's creating an in-memory state store instead of an S3 one.